lib/open-api-mocker.js

```diff
@@ -5,61 +5,90 @@ const jsonRefs = require('json-refs');
 const { Parser: OpenApiParser } = require('./openapi');
 const { Parser: ServersParser } = require('./servers');
 const { Parser: PathsParser } = require('./paths');
+const LocalSchemaLoader = require('./schema-loaders/local-loader');
+const OpenAPISchemaInvalid = require('./errors/openapi-schema-invalid-error');
+
+const optionsBuilder = require('./utils/options-builder');
+const ExplicitSchemaLoader = require('./schema-loaders/explicit-loader');
 
 class OpenApiMocker {
 
-	static get defaultPort() {
-		return 5000;
+	constructor(options) {
+		this.options = optionsBuilder(options);
+
+		if(this.options.schema)
+			this.setSchema(this.options.schema);
 	}
 
-	constructor({ port, server }) {
-		// Legacy handling of string based server that loads a server implementation based on folder name
-		if(typeof server === 'string' || server == null) {
-			// eslint-disable-next-line global-require, import/no-dynamic-require
-			const Server = require(`./mocker/${server || 'express'}/server`);
-			server = new Server();
+	setSchema(schema) {
+
+		if(this.schemaLoader && this.options.watch) {
+			if(this.schemaLoader.unwatch)
+				this.schemaLoader.unwatch();
+			this.schemaLoader.removeAllListeners();
+			this.options.alreadyWatching = false;
 		}
 
-		this.server = server;
-		this.port = port || this.constructor.defaultPort;
-	}
+		if(this.options.schemaLoader) {
+			const CustomSchemaLoader = this.options.schemaLoader;
+			this.schemaLoader = new CustomSchemaLoader(schema);
```

Comment:
Instead of `options.schemaLoader` being a class, would it be possible to make it an already instantiated instance?
If an instance was passed to the constructor it would give the calling code the ability to control when it gets instantiated and how long the variable lives for.
A new class what want to be configurable (i.e. a PollingSchemaLoader may want a configurable amount of time between each poll) but because the calling code has no control over when the schema loader gets instantiated, there is no way to control those configurable properties. This was the reasoning behind the change to allow a custom Server object rather than a custom Server class.
For this to work I guess that the instantiated object would require a `setSchema` method that would be called here instead of the constructor.